False positive: indentation of text block inside annotation array parameters

I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```cmd
E:\temp\CheckstyleReport>type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation" />
  </module>
</module>

E:\temp\CheckstyleReport>type Test.java
public class Test {
    @interface Anno {
        String [] value();
    }

    @Anno({
        """
        example
        """
    })
    void method() {
    }
}
E:\temp\CheckstyleReport>set RUN_LOCALE="-Duser.language=en -Duser.country=US"

E:\temp\CheckstyleReport>java %RUN_LOCALE% -jar checkstyle-13.0.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] E:\temp\CheckstyleReport\Test.java:9:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

**Describe what you expect in detail.**
End of text block formatting shouldn't cause violation of indentation rules.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AbstractExpressionHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AbstractExpressionHandler.java
@@ -330,7 +330,14 @@ public abstract class AbstractExpressionHandler {
             // checked by a child expression)
 
             if (col != null) {
-                checkLineIndent(astSet.getAst(index), level, false);
+                final DetailAST ast = astSet.getAst(index);
+                final boolean textBlockEndAligned =
+                    ast.getType() == TokenTypes.TEXT_BLOCK_LITERAL_END
+                        && isOnStartOfLine(ast.getParent())
+                        && indentLevel.isAcceptable(expandedTabsColumnNo(ast));
+                if (!textBlockEndAligned) {
+                    checkLineIndent(ast, level, false);
+                }
             }
         }
     }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -878,6 +878,62 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testAnnotationArrayInitWithTextBlockAfterLeftCurly() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(IndentationCheck.class);
+        checkConfig.addProperty("arrayInitIndent", "4");
+        checkConfig.addProperty("basicOffset", "4");
+        checkConfig.addProperty("braceAdjustment", "0");
+        checkConfig.addProperty("caseIndent", "4");
+        checkConfig.addProperty("throwsIndent", "4");
+        checkConfig.addProperty("lineWrappingIndentation", "4");
+        checkConfig.addProperty("tabWidth", "4");
+
+        final String[] expected = {
+            "38:5: " + getCheckMessage(MSG_CHILD_ERROR_MULTI,
+                "annotation array initialization", 4, "8, 26, 68"),
+            "47:5: " + getCheckMessage(MSG_CHILD_ERROR_MULTI,
+                "annotation array initialization", 4, "8, 26, 68"),
+            "49:5: " + getCheckMessage(MSG_CHILD_ERROR_MULTI,
+                "annotation array initialization", 4, "12, 30, 72"),
+            "79:5: " + getCheckMessage(MSG_CHILD_ERROR_MULTI,
+                "annotation array initialization", 4, "8, 12, 68"),
+            "81:5: " + getCheckMessage(MSG_CHILD_ERROR_MULTI,
+                "annotation array initialization", 4, "12, 16, 72"),
+            "89:5: " + getCheckMessage(MSG_CHILD_ERROR_MULTI,
+                "annotation array initialization", 4, "8, 12, 68"),
+        };
+        verifyWarns(checkConfig, getPath("InputIndentationAnnArrInitTextBlock.java"), expected);
+    }
+
+    @Test
+    public void testAnnotationArrayInitWithTextBlockAfterLeftCurly2() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(IndentationCheck.class);
+        checkConfig.addProperty("arrayInitIndent", "4");
+        checkConfig.addProperty("basicOffset", "4");
+        checkConfig.addProperty("braceAdjustment", "0");
+        checkConfig.addProperty("caseIndent", "4");
+        checkConfig.addProperty("throwsIndent", "4");
+        checkConfig.addProperty("lineWrappingIndentation", "4");
+        checkConfig.addProperty("tabWidth", "4");
+
+        final String[] expected = {
+            "13:13: " + getCheckMessage(MSG_CHILD_ERROR,
+                "annotation array initialization", 12, 8),
+            "17:13: " + getCheckMessage(MSG_CHILD_ERROR,
+                "annotation array initialization", 12, 8),
+            "21:13: " + getCheckMessage(MSG_CHILD_ERROR,
+                "annotation array initialization", 12, 8),
+            "37:13: " + getCheckMessage(MSG_CHILD_ERROR,
+                "annotation array initialization", 12, 8),
+            "39:5: " + getCheckMessage(MSG_CHILD_ERROR,
+                "annotation array initialization", 4, 12),
+        };
+        verifyWarns(checkConfig, getPath("InputIndentationAnnArrInitTextBlock2.java"), expected);
+    }
+
+    @Test
     public void testInvalidLabel() throws Exception {
         final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
 

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnArrInitTextBlock.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnArrInitTextBlock.java
@@ -1,0 +1,119 @@
+package                                                             //indent:0 exp:0
+    com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:4 exp:4
+
+public class InputIndentationAnnArrInitTextBlock {                  //indent:0 exp:0
+    @SuppressWarnings({                                             //indent:4 exp:4
+//below indent:8 exp:8
+        """
+        text block content                                          //indent:8 exp:8
+        on multiple lines                                           //indent:8 exp:8
+        """                                                         //indent:8 exp:8
+    })                                                              //indent:4 exp:4
+    public void method() {                                          //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+    @SuppressWarnings({                                             //indent:4 exp:4
+//below indent:8 exp:8
+        """
+        text block content                                          //indent:8 exp:8
+        on multiple lines                                           //indent:8 exp:8
+        with text block end                                         //indent:8 exp:8
+        not being at start                                          //indent:8 exp:8
+        of the line"""                                              //indent:8 exp:8
+    })                                                              //indent:4 exp:4
+    public void method1() {                                         //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+    @SuppressWarnings({                                             //indent:4 exp:4
+//below indent:8 exp:8
+        """
+        another text block                                          //indent:8 exp:8
+            """                                                     //indent:12 exp:12
+    })                                                              //indent:4 exp:4
+    public void method2() {                                         //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+    @SuppressWarnings({                                             //indent:4 exp:4
+//below indent:4 exp:8,26,68 warn
+    """
+    text block with wrong opening delimiter indent                  //indent:4 exp:4
+        """                                                         //indent:8 exp:8
+    })                                                              //indent:4 exp:4
+    public void method4() {                                         //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+    @SuppressWarnings({                                             //indent:4 exp:4
+//below indent:4 exp:8,26,68 warn
+    """
+    text block with wrong opening and closing delimiter             //indent:4 exp:4
+    """                                                             //indent:4 exp:12,30,72 warn
+    })                                                              //indent:4 exp:4
+    public void method5() {                                         //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+    @SuppressWarnings(                                              //indent:4 exp:4
+        {                                                           //indent:8 exp:8
+//below indent:12 exp:8,12
+            """
+            text block with brace on own line                       //indent:12 exp:12
+            and correct indentation                                 //indent:12 exp:12
+            """                                                     //indent:12 exp:8,12
+        })                                                          //indent:8 exp:8
+    public void method6() {                                         //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+    @SuppressWarnings(                                              //indent:4 exp:4
+        {                                                           //indent:8 exp:8
+//below indent:12 exp:8,12
+            """
+            text block with brace on own line                       //indent:12 exp:12
+            end delimiter extra indented                            //indent:12 exp:12
+                """                                                 //indent:16 exp:16
+        })                                                          //indent:8 exp:8
+    public void method7() {                                         //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+    @SuppressWarnings(                                              //indent:4 exp:4
+        {                                                           //indent:8 exp:8
+//below indent:4 exp:8,12,68 warn
+    """
+    brace on own line wrong opening indent                          //indent:4 exp:4
+    """                                                             //indent:4 exp:12,16,72 warn
+    })                                                              //indent:4 exp:4
+    public void method8() {                                         //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+    @SuppressWarnings(                                              //indent:4 exp:4
+        {                                                           //indent:8 exp:8
+//below indent:4 exp:8,12,68 warn
+    """
+    brace on own line wrong opening and closing                     //indent:4 exp:4
+        """                                                         //indent:8 exp:8,12
+    })                                                              //indent:4 exp:4
+    public void method9() {                                         //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+    @interface ApiParam {                                           //indent:4 exp:4
+        Example[] examples();                                       //indent:8 exp:8
+    }                                                               //indent:4 exp:4
+    @interface Example {                                            //indent:4 exp:4
+        ExampleProperty[] value();                                  //indent:8 exp:8
+    }                                                               //indent:4 exp:4
+    @interface ExampleProperty {                                    //indent:4 exp:4
+        String value() default "";                                  //indent:8 exp:8
+    }                                                               //indent:4 exp:4
+    @ApiParam(                                                      //indent:4 exp:4
+        examples = @Example(                                        //indent:8 exp:8
+            value = {                                               //indent:12 exp:12
+                @ExampleProperty(                                   //indent:16 exp:16
+//below indent:20 exp:20
+                    value = """
+                            generic text block content              //indent:28 exp:28
+                            on multiple lines                       //indent:28 exp:28
+                            """                                     //indent:28 exp:28
+                )                                                   //indent:16 exp:16
+            }                                                       //indent:12 exp:12
+        )                                                           //indent:8 exp:8
+    )                                                               //indent:4 exp:4
+    public void method10() {                                        //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+}                                                                   //indent:0 exp:0

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnArrInitTextBlock2.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnArrInitTextBlock2.java
@@ -1,0 +1,43 @@
+package                                                             //indent:0 exp:0
+    com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:4 exp:4
+
+public class InputIndentationAnnArrInitTextBlock2 {                 //indent:0 exp:0
+
+    @interface PropertySourceAnnotation {                           //indent:4 exp:4
+        String[] properties();                                      //indent:8 exp:8
+    }                                                               //indent:4 exp:4
+
+//below indent:4 exp:4
+    @PropertySourceAnnotation(properties = {
+//below indent:12 exp:8 warn
+            """
+            foo = bar                                               //indent:12 exp:12
+            """,                                                    //indent:12 exp:12
+//below indent:12 exp:8 warn
+            """
+            bar = baz                                               //indent:12 exp:12
+            """,                                                    //indent:12 exp:12
+//below indent:12 exp:8 warn
+            """
+            baz = quux                                              //indent:12 exp:12
+            """                                                     //indent:12 exp:12
+    })                                                              //indent:4 exp:4
+    public void method11() {                                        //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+//below indent:4 exp:4
+    @PropertySourceAnnotation(properties = {
+    })                                                              //indent:4 exp:4
+    public void method12() {                                        //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+
+//below indent:4 exp:4
+    @PropertySourceAnnotation(properties = {
+//below indent:12 exp:8 warn
+            "prefix" + """
+                    text content                                    //indent:20 exp:20
+    """                                                             //indent:4 exp:12 warn
+    })                                                              //indent:4 exp:4
+    public void method13() {                                        //indent:4 exp:4
+    }                                                               //indent:4 exp:4
+}                                                                   //indent:0 exp:0

